lib/cretonne/meta/cdsl/types.py

```diff
@@ -84,7 +84,10 @@ def __init__(self, name, membytes, doc):
         self._vectors = dict()  # type: Dict[int, VectorType]
         # Assign numbers starting from 1. (0 is VOID).
         ValueType.all_scalars.append(self)
-        self.number = len(ValueType.all_scalars)
+        # Numbers are only valid for Cretone types that get emitted to Rust.
+        # This excludes BVTypes
+        self.number = len([x for x in ValueType.all_scalars
+                           if not isinstance(x, BVType)])
```

Comment:
Perhaps it wold be better to make a whole new `ValueType` subclass for the bit vectors? They will need special treatment everywhere.

And something called "bit vector" doesn't strike me as belonging to "scalar types" ;-)